tests/kernel/fpu_sharing/float_disable/src/k_float_disable.c

```diff
@@ -148,7 +148,8 @@ void test_k_float_disable_syscall(void)
 #endif
 }
 
-#if defined(CONFIG_ARM) && defined(CONFIG_DYNAMIC_INTERRUPTS)
+#if defined(CONFIG_ARM) && defined(CONFIG_CPU_CORTEX_M) && \
+	defined(CONFIG_DYNAMIC_INTERRUPTS)
```

Comment:
> Cortex-R does not have an API for software to create a pending
interrupt like Cortex-M.  If we assume that the system is using a GIC
for the interrupt controller, it may be possible to create an ARM
specific API to generate an interrupt

It would be still useful to implement this testcase for `qemu_cortex_r5`, which uses the GICv1 as its interrupt controller.

Refer to the following for GIC-based interrupt generation:

https://github.com/zephyrproject-rtos/zephyr/blob/92f4db9fad1296a7e835f8dc167ce8aa8ea9657c/subsys/testsuite/include/interrupt_util.h#L73-L77

If you prefer to leave this to be done later, please add a `TODO: Implement GIC-based testcase.` here.

Reply:
Thanks.  I didn't realize that interface was there.  Updated the test.